lib/server/web.js

```diff
@@ -57,8 +57,8 @@ exports.create = function createServer() {
             token: token
           }
         }, function(err, resp, body) {
-          if (err) {
-            logger.error('auth', err);
+          if (err || resp.statusCode >= 500) {
+            err = err || resp.statusMessage || 'unknown';
             return reply(AppError.oauthError(err));
```

Comment:
It's because I was checking `body`, which I assumed was a json payload. If `err` is falsey, then `resp` **will** be a `http.IncomingMessage`. Just because we received a response doesn't mean the response had a body (the part after the headers), or that it was of content-type: json.

In this case, it was likely just `HTTP/1.1 500 Internal Server Error\r\nServer: ...\r\nDate: ...\r\n\r\n`, and so `body` was null.